dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/users/MergeInAccountService.java

```diff
@@ -0,0 +1,259 @@
+import edu.harvard.iq.dataverse.authorization.groups.ExplicitGroupDao;
+import edu.harvard.iq.dataverse.authorization.providers.builtin.BuiltinUserServiceBean;
+import edu.harvard.iq.dataverse.datafile.FileAccessRequestDao;
+import edu.harvard.iq.dataverse.guestbook.GuestbookResponseServiceBean;
+import edu.harvard.iq.dataverse.interceptors.LoggedCall;
+import edu.harvard.iq.dataverse.mail.confirmemail.ConfirmEmailServiceBean;
+import edu.harvard.iq.dataverse.persistence.DvObject;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetLock;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersionUser;
+import edu.harvard.iq.dataverse.persistence.dataverse.link.SavedSearch;
+import edu.harvard.iq.dataverse.persistence.guestbook.GuestbookResponse;
+import edu.harvard.iq.dataverse.persistence.user.ApiToken;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUserLookup;
+import edu.harvard.iq.dataverse.persistence.user.BuiltinUser;
+import edu.harvard.iq.dataverse.persistence.user.ConfirmEmailData;
+import edu.harvard.iq.dataverse.persistence.user.RoleAssignment;
+import edu.harvard.iq.dataverse.persistence.user.User;
+import edu.harvard.iq.dataverse.persistence.user.UserNotification;
+import edu.harvard.iq.dataverse.persistence.user.UserNotificationDao;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowComment;
+import edu.harvard.iq.dataverse.search.index.IndexServiceBean;
+import edu.harvard.iq.dataverse.search.index.SolrIndexServiceBean;
+import edu.harvard.iq.dataverse.search.savedsearch.SavedSearchServiceBean;
+
+import javax.ejb.EJB;
+import javax.ejb.Stateless;
+import java.util.List;
+import java.util.Optional;
+
+@Stateless
+public class MergeInAccountService {
+
+    @EJB private AuthenticationServiceBean authenticationService;
+    @EJB private RoleAssigneeServiceBean roleAssigneeService;
+    @EJB private SolrIndexServiceBean solrIndexService;
+    @EJB private IndexServiceBean indexService;
+    @EJB private DatasetDao datasetDao;
+    @EJB private DvObjectServiceBean dvObjectService;
+    @EJB private GuestbookResponseServiceBean guestbookResponseService;
+    @EJB private UserNotificationDao userNotificationDao;
+    @EJB private SavedSearchServiceBean savedSearchService;
+    @EJB private ConfirmEmailServiceBean confirmEmailService;
+    @EJB private BuiltinUserServiceBean builtinUserService;
+    @EJB private GenericDao genericDao;
+    @EJB private FileAccessRequestDao fileAccessRequestDao;
+    @EJB private ExplicitGroupDao explicitGroupDao;
+    @EJB private AcceptedConsentDao acceptedConsentDao;
+    @EJB private OAuthTokenDataDao oAuthTokenDataDao;
+
+    // -------------------- LOGIC --------------------
+
+    @LoggedCall
+    public void mergeAccounts(User user, String consumedIdentifier, String baseIdentifier)
+        throws IllegalArgumentException, SecurityException {
```

Comment:
Also that now I noticed why are you using RuntimeExceptions as checked exceptions? They are meant not to be used as such, that's why they are runtime :P.